Adding responsive web page with bootstrap 4

kots14 opened the pull request:
**Issue:** https://github.com/KamandPrompt/baat-cheet/issues/66#issue-713835681

Added responsive home page with some style changes to the home page. Also tried to arrange the ejs templates for the project.

**Summary of changes**

- `container` class in style.css renamed to `app-container` in order to avoid conflict with bootstrap's `container` class
- Separated the headers and footers in `index.ejs` and added bootstrap imports in them
- Added some code in style.css to style the home page

**Note :** 

- [x] Tested responsiveness of home page in Google Chrome
- [x] Tested responsiveness of home page in Mozilla Firefox

**Update:**
- As mentioned [here](https://github.com/KamandPrompt/baat-cheet/pull/70#issuecomment-703033530), fixed the bugs in the PR. Renamed `container` to `app-container` in `client.js` to fix the issue.

abhigyank committed:
Re-add "SHOW ONLINE USERS"" (#57)

* Revert "Revert "SHOW ONLINE USERS now one can click on user-count(under the room name) to see name of online users""

This reverts commit 0d4f33534ff10cb80c26f021270db73a96decd2d.

* Revert "Revert "Participants list: ellipsis in case of long name.""

This reverts commit fd8288da7d3f2f12c0d92887ff0a9c0deae27dee.

LalitNM committed:
Added collapsible sidebar (#54)

divyashk committed:
Updated the CDN for socket.io (#59)

Didn't change the version to avoid incompatibilities, just updated the address.

Solai-Adithya committed:
Close sidebar when clicked outside sidebar (#63)

Close sidebar when clicked outside sidebar

kots14 committed:
Adding responsive web page with bootstrap 4

Added responsive home page with some style changes to the home page. Also tried to arrange the ejs templates for the project.

LalitNM commented:
I would like to point out some bugs, which this PR creates:

- After adding a new room, the screen overlay is not removed automatically.
steps to reproduce: try adding a new room
- The top penal in any new room is not displayed.
steps to reproduce: add a new room and look for top penal(which includes the room name and information about online users)

In addition to these you need to look again the responsiveness of entry page, you need to improve this.

kots14 commented:
I'm trying to fix these bugs. I'll get back to you if I need any help.

kots14 committed:
Merge branch 'master' of https://github.com/kots14/baat-cheet

kots14 committed:
Adding responsive home page using bootstrap 4

fixing bugs as mentioned in https://github.com/KamandPrompt/baat-cheet/pull/70#issuecomment-703033530

kots14 commented:
I've updated the PR and issues seems to be fixed. @LalitNM please let me know if there's anything I missed in the PR that has to be worked on.

LalitNM commented:
This PR can be further improved by

- removing unnecessary styling code from `style.css`.

- adding a `text-center` class to the `#user` division.

- adding a `font-size` attribute to the error message which comes when two users try to enter with same name.

@kots14 when you will do the above changes, we can merge this.

kots14 commented:
> This PR can be further improved by
> 
> * removing unnecessary styling code from `style.css`.
> * adding a `text-center` class to the `#user` division.
> * adding a `font-size` attribute to the error message which comes when two users try to enter with same name.
> 
> @kots14 when you will do the above changes, we can merge this.

@LalitNM First and Second requirements done. I'm confused on the 3rd requirement. Currently the `font-size` for `#error_response` is `1.3vw`. Shall I change it to a new value?

kots14 committed:
Adding changes to home page using bootstrap 4

Trying to make changes as mentioned here https://github.com/KamandPrompt/baat-cheet/pull/70#issuecomment-703102746

LalitNM commented:
> > This PR can be further improved by
> > 
> > * removing unnecessary styling code from `style.css`.
> > * adding a `text-center` class to the `#user` division.
> > * adding a `font-size` attribute to the error message which comes when two users try to enter with same name.
> > 
> > @kots14 when you will do the above changes, we can merge this.
> 
> @LalitNM First and Second requirements done. I'm confused on the 3rd requirement. Currently the `font-size` for `#error_response` is `1.3vw`. Shall I change it to a new value?

Change that in `rem` units, and decide which value you want to give it to, maybe 1 or 0.5 can work.

kots14 committed:
Changed font size for #error_response

kots14 commented:
Updated the `font-size` to 1rem for the `#error_response`.
@LalitNM Let me know if there's any other scope of improvement in this PR.

LalitNM commented:
> @LalitNM Let me know if there's any other scope of improvement in this PR.

 Few more changes

- change `font-size` of title.

- add jumbotron again which you have removed in last commit.

kots14 commented:
For Jumbotron what's the background color and foreground color you'd like? Shall I keep it default?
Also for title is this the snippet you expect me to change?
```
 #bc {
    font-size: 16vw !important;
  }
```
What's the expected font size?

LalitNM commented on public/style.css:
```diff
@@ -25,11 +25,6 @@ li {
-  background: linear-gradient(#000,  #00b0ff);
-  border: 2px solid  #00b0ff;
-}
-
```
@kots14 this is what I am talking about for jumbotron, you can add this code in your next commit.

kots14 replied:
got it.

LalitNM commented:
> #bc {
>    font-size: 16vw !important;
>  }
>
>What's the expected font size?

You can add font-size in `rem` and check which font-size looks good, I just want to remove font-size based on viewport-width.

kots14 committed:
Adding jumbotron based styles, adding new font size for title (#bc)

kots14 commented:
@LalitNM I've updated the PR. You can check and let me know if there are any other changes required.

LalitNM commented:
Merged :rocket: Good work @kots14.

abhigyank commented on public/style.css:
```diff
@@ -25,6 +25,11 @@ li {
 }
 
+#user .home-label {
+  background: linear-gradient(#000,  #00b0ff);
```
I dont suppose this blue gradient is looking good. Doesn't the pure black look better?

LalitNM replied:
Pure black was fine but I thought that we should highlight the useful area by making a blue gradient.

abhigyank replied:
Okay, understood. In my opinion, doesn't look very aesthetically pleasing.

kots14 replied:
With the current file, I can make it look like option 1 or 2 with a small commit. I can raise a new Issue on this repository and make this change according to that.

1. ![image](https://user-images.githubusercontent.com/20207952/94999820-b2c18080-05d9-11eb-9ff8-c51026f80bc6.png)
2.  ![image](https://user-images.githubusercontent.com/20207952/94999717-f23b9d00-05d8-11eb-9189-14be5245f1e8.png)

However, if it doesn't fit the requirement, please let me know when you raise a new Issue specifying the required look for this. I'd like to work on that.

abhigyank replied:
@kots14 Let's go with 2. Since Lalit wanted to insert a blue shade. Can you make a PR for this?

LalitNM replied:
One more problem this PR creates is the git history. After merging this PR the git history is clumsy. I am gonna revert all changes done in this PR and will merge your next PR with the changes requested by @abhigyank and these changes. Make sure to that you have added all these changes in new PR. If you need the present state of repository for any reference then check [70-merged](https://github.com/KamandPrompt/baat-cheet/tree/70-merged) branch @kots14.

kots14 replied:
@LalitNM Thanks for creating the reference branch. I'll raise a new PR with the changes mentioned by @abhigyank.

kots14 replied:
Added new PR https://github.com/KamandPrompt/baat-cheet/pull/82 for https://github.com/KamandPrompt/baat-cheet/issues/66
